dbms/src/Flash/Coprocessor/IChunkDecodeAndSquash.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Coprocessor/IChunkDecodeAndSquash.h>
+#include <IO/ReadBufferFromString.h>
+
+namespace DB
+{
+CHBlockChunkDecodeAndSquash::CHBlockChunkDecodeAndSquash(
+    const Block & header,
+    size_t rows_limit_)
+    : codec(header)
+    , rows_limit(rows_limit_)
+{
+}
+
+std::optional<Block> CHBlockChunkDecodeAndSquash::decodeAndSquash(const String & str)
```

Comment:
It seems to be fine to use `Block`.
https://github.com/pingcap/tiflash/blob/8b123a83aa6150768b746d089f73dd8205314cc8/dbms/src/Core/Block.h#L113-L114

Reply:
There's a little differences between block's bool and decodeAndSquash return value:
Block's bool just indicate whether there the block has column info, usually used to check if it is the end of a stream. And other corner case might produce it also.
optional value just make it more clear.

Reply:
ok